catalog/registry/catalog-registry-api-impl/src/main/java/ddf/catalog/registry/api/impl/RegistryStoreImpl.java

```diff
@@ -0,0 +1,207 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.registry.api.impl;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+
+import javax.xml.bind.JAXBElement;
+import javax.xml.bind.Marshaller;
+
+import org.codice.ddf.cxf.SecureCxfClientFactory;
+import org.codice.ddf.parser.Parser;
+import org.codice.ddf.parser.ParserConfigurator;
+import org.codice.ddf.parser.ParserException;
+import org.codice.ddf.spatial.ogc.csw.catalog.common.CswSourceConfiguration;
+import org.codice.ddf.spatial.ogc.csw.catalog.common.source.AbstractCswSource;
+import org.osgi.framework.BundleContext;
+
+import com.google.common.base.Charsets;
+import com.google.common.io.CharStreams;
+import com.google.common.io.FileBackedOutputStream;
+import com.thoughtworks.xstream.converters.Converter;
+
+import ddf.catalog.Constants;
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.operation.OperationTransaction;
+import ddf.catalog.operation.UpdateRequest;
+import ddf.catalog.operation.UpdateResponse;
+import ddf.catalog.registry.api.RegistryStore;
+import ddf.catalog.registry.api.metacard.RegistryObjectMetacardType;
+import ddf.catalog.registry.common.RegistryConstants;
+import ddf.catalog.source.IngestException;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.ExternalIdentifierType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.RegistryObjectType;
+
+public class RegistryStoreImpl extends AbstractCswSource implements RegistryStore {
+
+    public static final String PUBLISH_ENABLED_PROPERTY = "publishEnabled";
```

Comment:
Consider renaming these to match what we want to see on the UI (allow push, allow pull)